### PR TITLE
Support Heart Rate sensor (PT-186280507)

### PR DESCRIPTION
**Pull request by lublagg**
This PR adds support for the Heart Rate sensor option of the GDX-EKG sensor by manually checking if we are using this sensor and then passing slower values to `this.gdxDevice.start`.

### CODE REVIEW

**Review by scytacki — APPROVED** [review]
Looks good to me.

Just for reference incase someone looks back at this PR:
When the heart rate sensor is configured with a period less than 1s, it returns NaN's for the values between the 1s readings. The sensor-gdx-manager is currently ignoring the `sensor.values` array and only using the `sensor.value`.  This `sensor.value` is the last read value; `sensor.values` includes all of the read values. So with NaNs, it is often the case that `sensor.value` reports a NaN even though there was an valid value earlier in the values array.

File: src/models/sensor-gdx-manager.ts [hunk 1]
```diff
@@ -123,10 +118,38 @@ export class SensorGDXManager extends SensorManager {
...
     }
 
+    // This value gets sent to gdxDevice.start() to set the measurement period.
+    // It gets used when we've actually started the sensor data graphing.
+    getMeasurementPeriod() {
+      // The Heart Rate sensor is only able to collect 1 sample per second.
+      // Currently HEARTBEAT_INTERVAL_MS is set to 1000ms (1 second).
+      // If that changes we'll need to manually return 1000 here.
+      if (this.gdxDevice.orderCode === "GDX-EKG") {
+        return HEARTBEAT_INTERVAL_MS;
```
Comment (dougmartin) [inline src/models/sensor-gdx-manager.ts:128]: I think you should go ahead and just return a new constant of `MIN_EKG_INTERVAL_MS` (set to 1000) here (with a comment of why it is 1000 where you create the constant) so if someone does change `HEARTBEAT_INTERVAL_MS` in the future without seeing this comment it doesn't break the sensor.

File: src/models/sensor-gdx-manager.ts [hunk 2]
```diff
@@ -123,10 +118,38 @@ export class SensorGDXManager extends SensorManager {
...
+        return this.gdxDevice.minMeasurementPeriod;
+      }
+    }
+
+    // This value also gets sent to gdxDevice.start() to set the measurement period.
+    // It gets used when we're displaying the calibration values, not during collecting / graphing.
+    getHeartbeatInterval() {
+      if (this.gdxDevice.orderCode === "GDX-EKG") {
+        // See notes above in getMeasurementPeriod().
+        return HEARTBEAT_INTERVAL_MS;
```
Comment (dougmartin) [inline src/models/sensor-gdx-manager.ts:139]: Same comment as above, I think this should change to the new `MIN_EKG_INTERVAL_MS` constant.

**Review by dougmartin — APPROVED** [review]
Looks good.  Just a suggestion to future proof the code in my comments.

**Pull request merged**